src/polkit/polkitunixprocess.c

```diff
@@ -376,37 +376,19 @@ polkit_unix_process_constructed (GObject *object)
 {
   PolkitUnixProcess *process = POLKIT_UNIX_PROCESS (object);
 
-  /* sets pidfd, start_time and uid in case they are unset */
+  /* sets start_time and uid in case they are unset */
 
   /* We didn't open it ourselves here, so we must have got it
    * from D-Bus, mark it as safe to use */
   if (process->pidfd >= 0)
     process->pidfd_is_safe = TRUE;
 
 #ifdef HAVE_PIDFD_OPEN
-  if (process->pid > 0 && process->pidfd < 0)
```

Comment:
This doesn't look right, it depends on how the object is created whether it's set or not. There's a check to see if it was already set, so just leave this as-is